Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -2430,6 +2430,64 @@ bool Element::check_visibility(Optional<CheckVisibilityOptions> options)
     return true;
 }
 
+// https://drafts.csswg.org/css-contain/#proximity-to-the-viewport
+void Element::determine_proximity_to_the_viewport()
+{
+    // An element that has content-visibility: auto is in one of three states when it comes to its proximity to the viewport:
+
+    // - The element is close to the viewport: In this state, the element is considered "on-screen": its paint
+    //   containment box's overflow clip edge intersects with the viewport, or a user-agent defined margin around the
+    //   viewport.
+    auto viewport_rect = document().viewport_rect();
+    // NOTE: This margin is meant to allow the user agent to begin preparing for an element to be in the
+    // viewport soon. A margin of 50% is suggested as a reasonable default.
+    viewport_rect.inflate(viewport_rect.width(), viewport_rect.height());
+    // FIXME: We don't have paint containment or the overflop clip edge yet, so this is just using the absolute rect for now.
```

Comment:
Typo: "overflop". (Though I love this as a word!)

Reply:
Good point, fixed.